package-res/reportviewer/reportviewer-prompt.js

```diff
@@ -73,10 +75,140 @@ define([
         return this.api.operation.state()[prop];
       },
 
+      isParameterUpdateCall: function() {
+        // never had an old definition, so we cant optimize anything.
+        if (this._oldParameterDefinition == null) {
+          return false;
+        }
+        return true;
+      },
+
+      compareParameters: function(oldParameterSet, currentParameterSet) {
+        var changedParameters = [];
+
+        $.each(oldParameterSet, function (i, parametersGroup) {
+          if("parameters" == parametersGroup.group) {
```

Comment:
Please make this test reverse: Test for "system" as group-name. This is a well-defined hard-coded name encapsulating all system level parameters like output-target, scheduling etc. This way you leave it open for other implementations to define additional parameter-groups. "parameters" is not one of the prescribed names, it is more or less just an implementation detail of the current level of support.